week04/code/LinkedListTester.cs

```diff
@@ -16,12 +16,24 @@ public static class LinkedListTester {
         ll.InsertTail(-1);
         Console.WriteLine(ll.ToString()); // <LinkedList>{5, 4, 3, 2, 2, 2, 1, 0, -1};
 
+        var ll2 = new LinkedList();
+        ll2.InsertTail(1);
+        Console.WriteLine(ll2.HeadAndTailAreNotNull()); // True
+
         Console.WriteLine("\n=========== PROBLEM 2 TESTS ===========");
         ll.RemoveTail();
         Console.WriteLine(ll.ToString()); // <LinkedList>{5, 4, 3, 2, 2, 2, 1, 0}
         ll.RemoveTail();
         Console.WriteLine(ll.ToString()); // <LinkedList>{5, 4, 3, 2, 2, 2, 1}
 
+        var ll3 = new LinkedList();
+        ll3.RemoveTail();
+        Console.WriteLine(ll3.ToString()); // <LinkedList>{}
+        ll3.InsertHead(2);
+        ll3.RemoveTail();
+        Console.WriteLine(ll3.ToString()); // <LinkedList>{}
+        Console.WriteLine(ll3.HeadAndTailAreNull()); // True
+
```

Comment:
Catches `RemoveTail` not working properly on an empty list or a list with one element.